Tasks/ServiceFabricComposeDeploy/ServiceFabricComposeDeploy.ps1

```diff
@@ -22,25 +22,68 @@ try {
     $removeTimeoutSec = Get-VstsInput -Name removeTimeoutSec
     $getStatusTimeoutSec = Get-VstsInput -Name getStatusTimeoutSec
 
-    $deployParameters = @{
-        'ApplicationName' = $applicationName
-        'Compose' = $composeFilePath
-    }
-    $removeParameters = @{
-        'Force' = $true
-    }
-    $getStatusParameters = @{
-        'ApplicationName' = $applicationName
-    }
-
-    $usePreviewApi = $false
-    $regKey = Get-ItemProperty -Path 'HKLM:\SOFTWARE\Microsoft\Service Fabric\' -ErrorAction SilentlyContinue
+    $apiVersion = '2.8'
+    $regKey = Get-ItemProperty -Path 'HKLM:\SOFTWARE\Microsoft\Service Fabric SDK\' -ErrorAction SilentlyContinue
     if ($regKey)
     {
-        if ($regKey.FabricVersion.StartsWith('255.255'))
+        if ($regKey.FabricSDKVersion -match "^\d+\.\d+")
+        {
+            $regExVersion = $matches[0]
+
+            switch ($regExVersion) {
+                '2.7' { $apiVersion = '2.7' }
+                '2.8' { $apiVersion = '2.8' }
+                '255.255' { $apiVersion = '255.255' }
+                Default {
+                    $sdkVersion = New-Object Version
+                    if ([Version]::TryParse($matches[0], [ref]$sdkVersion)) {
+                        $minVersion = New-Object -TypeName Version -ArgumentList '2.7'
+                        if ($sdkVersion -lt $minVersion) {
```

Comment:
In else case, should we update the ApiVersion to SdkVersion - or do we want to default to 2.8?

Reply:
I wanted to leave it as 2.8 as that is the highest known version of the API. If there is an incompatible change in the future more cases will need to be added.

Reply:
Ok